test/res.send.js

```diff
@@ -250,6 +250,77 @@ describe('res', function(){
     })
   })
 
+  describe('.send(Blob)', function(){
+    var Blob = require('buffer').Blob;
+
+    it('should send as blob type', function (done) {
+      if (Blob) {
+        var str = '<h1>express app</h1>';
+        var blob = new Blob([str], { type: 'text/html' });
+        var app = express();
+        app.use(function (req, res) {
+          res.send(blob);
+        });
+
+        request(app)
+        .get('/')
+        .expect('Content-Type','text/html; charset=utf-8')
```

Comment:
I would not have assumed that it would append `charset=utf-8`

that's not what happens if you would do it with `Response()`

```js
onfetch = evt => evt.respondWith(
  new Response(
    new Blob(['<p>'], { type: 'text/html' })
  ) // .headers.get('content-type') === 'text/html'
)
```

Reply:
I agree. The `Blob` is just a byte stream, so whatever the `type` is shouldn't be touched, as Express has no way to know what the chatset is of the bytestream.

Reply:
Another thing that the `XMLHttpRequest` and `fetch` api dose is that you are able to override the `content-type` header. 

`content-type` is only added from `blob`s type if `content-type` haven't been set

so by doing this:
```js
const blob = new Blob([], { type: 'text/plain '})
const xhr = new XMLHttpRequest()
xhr.setRequestHeader('content-type', 'text/html')
xhr.open('POST', url)
xhr.send()
```
...would result in a `content-type` being set to `text/html` (not the blob's own type)
two `content-type` headers would not have been sent.

So with that said i would like to also mention that if somebody where to write / set a `content-type` header themself. then express should not try to add content-type header originating from the blob's own `type`

Reply:
Makes sense. I think that is what this pr is already doing, but if not, it shouldn't set the type if already set, which is what all the other arguments to send do.

Reply:
> Makes sense. I think that is what this pr is already doing, but if not, it shouldn't set the type if already set, which is what all the other arguments to send do.

@dougwilson Yes, it is already there

Reply:
I'm not sure why this was marked resolved, as nothing was fixed and the issue @jimmywarting mentioned with adding charset is still there.